src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -273,6 +273,26 @@ public void UpdateDataSourceNameToDataSource(string dataSourceName, DataSource d
         _dataSourceNameToDataSource[dataSourceName] = dataSource;
     }
 
+    /// <summary>
+    /// In a Hot Reload scenario we should maintain the same default data source
+    /// name before the hot reload as after the hot reload. This method takes
+    /// a default data source name, such as the one from before hot reload, and
+    /// replaces the current dictionary entries of this RuntimeConfig that were
+    /// built using a new, unique guid during the construction of this RuntimeConfig
+    /// with entries using the provided default data source name.
+    /// </summary>
+    /// <param name="defaultDataSourceName">The name used to update the dictionaries.</param>
+    public void UpdateDefaultDataSourceNameDependantDictionaries(string defaultDataSourceName)
+    {
+        _dataSourceNameToDataSource.Remove(DefaultDataSourceName);
+        _dataSourceNameToDataSource.Add(defaultDataSourceName, this.DataSource);
+        foreach (KeyValuePair<string, Entity> entity in Entities)
+        {
+            _entityNameToDataSourceName.Remove(entity.Key);
+            _entityNameToDataSourceName.TryAdd(entity.Key, DefaultDataSourceName);
```

Comment:
You are removing the entity.key but readding it with the old value here. Did you test this?

Reply:
i did but I think I got the names mixed up, I renamed the argument to try and be more clear about which data source names are which.